package/MDAnalysis/coordinates/memory.py

```diff
@@ -0,0 +1,242 @@
+
+    def _reopen(self):
+        """Reset iteration to first frame"""
+        self.ts.frame = -1
+
+    def timeseries(self, asel=None, start=0, stop=-1, skip=1, format='afc'):
+        """Return a subset of coordinate data for an AtomGroup in desired
+        column format. If no selection is given, it will return a view of
+        the underlying array, while a copy is returned otherwise.
+
+        Parameters
+        ---------
+        asel : :class:`~MDAnalysis.core.AtomGroup.AtomGroup` object
+            Atom selection
+        start, stop, skip : int
+            range of trajectory to access, start and stop are inclusive
+        format : str
+            the order/shape of the return data array, corresponding
+            to (a)tom, (f)rame, (c)oordinates all six combinations
+            of 'a', 'f', 'c' are allowed ie "fac" - return array
+            where the shape is (frame, number of atoms,
+            coordinates)
+        """
+        array = self.get_array()
+        if format == self.format:
+            pass
+        elif format[0] == self.format[0]:
+            array = np.swapaxes(array, 1, 2)
+        elif format[1] == self.format[1]:
+            array = np.swapaxes(array, 0, 2)
+        elif format[2] == self.format[2]:
+            array = np.swapaxes(array, 0, 1)
+        elif self.format[1] == format[0]:
+            array = np.swapaxes(array, 1, 0)
+            array = np.swapaxes(array, 1, 2)
+        elif self.format[2] == format[0]:
+            array = np.swapaxes(array, 2, 0)
+            array = np.swapaxes(array, 1, 2)
+
+        a_index = format.find('a')
+        f_index = format.find('f')
+        stop_index = stop+1
+        if stop_index == 0:
+            stop_index = None
+        basic_slice = ([slice(None)]*(f_index) +
+                       [slice(start, stop_index, skip)] +
+                       [slice(None)]*(2-f_index))
+
+        # Return a view if either:
+        #   1) asel is None
+        #   2) asel corresponds to the selection of all atoms.
+        array = array[basic_slice]
+        if (asel is None or asel is asel.universe.atoms):
+            return array
+        else:
+            # If selection is specified, return a copy
+            return array.take(asel.indices, a_index)
+
+    def _read_next_timestep(self, ts=None):
+        """copy next frame into timestep"""
+
+        if self.ts.frame >= self.n_frames-1:
+            raise IOError(errno.EIO, 'trying to go over trajectory limit')
+        if ts is None:
+            ts = self.ts
+        ts.frame += 1
+        f_index = self.format.find('f')
+        basic_slice = ([slice(None)]*(f_index) +
+                       [self.ts.frame] +
+                       [slice(None)]*(2-f_index))
+        ts.positions = self.coordinate_array[basic_slice]
+
+        ts.time = self.ts.frame*self.dt
+        return ts
+
+    def _read_frame(self, i):
+        """read frame i"""
+        self.ts.frame = i-1
```

Comment:
A quick comment on why the `- 1` would be neat here (just saying that `_read_next_timestep` increment `self.ts.frame`).

Also, spaces around the `-`.